src/material-experimental/mdc-chips/chip-icons.ts

```diff
@@ -0,0 +1,137 @@
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ChangeDetectorRef,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Output,
+} from '@angular/core';
+import {
+  CanDisable,
+  CanDisableCtor,
+  HasTabIndex,
+  HasTabIndexCtor,
+  mixinDisabled,
+  mixinTabIndex,
+} from '@angular/material/core';
+import {FocusableCell} from './grid-focus-key-manager';
+
+
+/**
+ * Directive to add CSS classes to chip leading icon.
+ * @docs-private
+ */
+@Directive({
+  selector: 'mat-chip-avatar, [matChipAvatar]',
+  host: {
+    'class': 'mat-mdc-chip-avatar mdc-chip__icon mdc-chip__icon--leading',
+    'role': 'img'
+  }
+})
+export class MatChipAvatar {
+  constructor(private _changeDetectorRef: ChangeDetectorRef,
+              private _elementRef: ElementRef) {}
+
+  /** Sets whether the given CSS class should be applied to the leading icon. */
+  setClass(cssClass: string, active: boolean) {
+    const element = this._elementRef.nativeElement;
+    active ? element.addClass(cssClass) : element.removeClass(cssClass);
+    this._changeDetectorRef.markForCheck();
+  }
+}
+
+/**
+ * Directive to add CSS classes to and configure attributes for chip trailing icon.
+ * @docs-private
+ */
+@Directive({
+  selector: 'mat-chip-trailing-icon, [matChipTrailingIcon]',
+  host: {
+    'class': 'mat-mdc-chip-trailing-icon mdc-chip__icon mdc-chip__icon--trailing',
+    '[tabIndex]': '-1',
+    'aria-hidden': 'true',
+  }
+})
+export class MatChipTrailingIcon {
+}
+
+/**
+ * Boilerplate for applying mixins to MatChipRemove.
+ * @docs-private
+ */
+class MatChipRemoveBase extends MatChipTrailingIcon {
+  constructor(public _elementRef: ElementRef) {
+    super();
+  }
+}
+
+const _MatChipRemoveMixinBase:
+  CanDisableCtor &
+  HasTabIndexCtor &
+  typeof MatChipRemoveBase =
+    mixinTabIndex(mixinDisabled(MatChipRemoveBase));
+
+/**
+ * Directive to remove the parent chip when the trailing icon is clicked or
+ * when the ENTER key is pressed on it.
+ *
+ * Recommended for use with the Material Design "cancel" icon
+ * available at https://material.io/icons/#ic_cancel.
+ *
+ * Example:
+ *
+ *     `<mat-chip>
+ *       <mat-icon matChipRemove>cancel</mat-icon>
+ *     </mat-chip>`
+ */
+@Directive({
+  selector: '[matChipRemove]',
+  inputs: ['disabled', 'tabIndex'],
+  host: {
+    'class': 'mat-mdc-chip-trailing-icon mdc-chip__icon mdc-chip__icon--trailing',
+    '[tabIndex]': 'tabIndex',
+    'role': 'button',
+    '(click)': 'interaction.emit($event)',
+    '(keydown)': 'keydown($event)',
+    '(blur)': 'blur()',
+    '(focus)': 'focus()'
+  }
+})
+export class MatChipRemove extends _MatChipRemoveMixinBase implements CanDisable, FocusableCell,
+  HasTabIndex {
+  _hasFocus = false;
+
+  /** Emits when the user interacts with the icon. */
+  @Output() interaction = new EventEmitter<MouseEvent | KeyboardEvent>();
```

Comment:
Could this be marked as internal and also just be a plain subject without `@Output`? The decorator is only necessary fi you want to define an event handler in a template